plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/visitors/NoSelfChecker.java

```diff
@@ -176,6 +177,22 @@ public void beforeFunctionDef(FunctionDef node) {
         scope.push(Scope.SCOPE_TYPE_METHOD);
     }
 
+    private static boolean hasInterfaceInBase(FunctionDef node) {
```

Comment:
Will this method work for classes that don't directly inherit from `zope.interface.Interface`? For example:

```python
from zope.interface import Interface

class Ia(Interface):
    def method_a():  # Shouldn't complain about the lack of self
        """"""

class Ib(Ia):
    def method_b():  # Shouldn't complain about the lack of self
        """"""

```

Reply:
This doesn't work right now, but it should (so, we should have a test-case for this use-case).